### PR TITLE
[PA-818] Manual and schedule backup performed on single, multiple and all namespace using namespace label with max chars

### PR DESCRIPTION
**Pull request by apimpalgaonkar**
**What this PR does / why we need it**:
In this PR, multiple namespaces are created with namespace label of max 63 character
Now manual backup and schedule backup is performed on single , multiple and all namespaces.
Restore operation on same.

**Which issue(s) this PR fixes** (optional)
Closes #
https://portworx.atlassian.net/browse/PA-818

Jenkins - 
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1088/consoleFull

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_restore_basic_test.go [hunk 1]
```diff
@@ -1318,3 +1319,310 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
...
+			log.InfoD("Taking manual backup of all applications with namespace label filter")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			manualBkpAllNs = fmt.Sprintf("%s-%v", "all-namespace-backup", time.Now().Unix())
+			err = CreateBackupWithNamespaceLabel(manualBkpAllNs, SourceClusterName, backupLocationName, backupLocationUID,
+				labelSelectors, orgID, clusterUid, "", "", "", "", labelForAllNamespace, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup [%s] creation with labels [%v]", manualBkpAllNs, labelForAllNamespace))
+			err = NamespaceLabelBackupSuccessCheck(manualBkpAllNs, ctx, bkpNamespaces, labelForAllNamespace)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if the labeled namespaces [%v] are backed up and check for labels [%s] applied to backups [%s]", bkpNamespaces, labelForAllNamespace, manualBkpAllNs))
+		})
```
Comment (skonda-px) [inline tests/backup/backup_restore_basic_test.go:1495]: In all three scenarios most of the code is repeated only. You can write a common method where you can pass the labels and applied labels namespaces and method will return the backupname.

File: tests/backup/backup_restore_basic_test.go [hunk 2]
```diff
@@ -1318,3 +1319,310 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
...
+			scheduleBkpAllNs = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+			err = CreateScheduleBackupWithNamespaceLabel(scheduleBkpAllNs, SourceClusterName, backupLocationName, backupLocationUID,
+				labelSelectors, orgID, "", "", "", "", labelForAllNamespace, periodicSchedulePolicyName, periodicSchedulePolicyUid, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup [%s] with labels [%v]", scheduleBkpAllNs, labelForAllNamespace))
+			firstScheduleBackupForAllNs, err = GetFirstScheduleBackupName(ctx, scheduleBkpAllNs, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup [%s]", firstScheduleBackupForAllNs))
+			err = NamespaceLabelBackupSuccessCheck(firstScheduleBackupForAllNs, ctx, bkpNamespaces, labelForAllNamespace)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if the labeled namespaces [%v] are backed up and check for labels [%s] applied to backups [%s]", bkpNamespaces, labelForAllNamespace, firstScheduleBackupForAllNs))
+			scheduleNames = append(scheduleNames, scheduleBkpAllNs)
+		})
```
Comment (skonda-px) [inline tests/backup/backup_restore_basic_test.go:1549]: Here also same you can write a common method.

File: tests/backup/backup_restore_basic_test.go [hunk 3]
```diff
@@ -1318,3 +1319,310 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
...
+			err = CreateRestore(restoreName, manualBkpAllNs, nil, destinationClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying all backups [%s] restore in default namespace", restoreName))
+			restoreNames = append(restoreNames, restoreName)
+		})
+		Step("Restoring first scheduled backup of single namespace", func() {
+			log.InfoD("Restoring first scheduled backup of single namespace")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			restoreName = fmt.Sprintf("%s-%v", restoreNamePrefix, time.Now().Unix())
+			err = CreateRestore(restoreName, firstScheduleBackupName, nil, destinationClusterName, orgID, ctx, nil)
```
Comment (skonda-px) [inline tests/backup/backup_restore_basic_test.go:1582]: Since for manual backups tried the default restore, try custom restores for scheduled backups

**Review by skonda-px — CHANGES_REQUESTED** [review]

File: tests/backup/backup_restore_basic_test.go (hunk 3 above)
Comment (apimpalgaonkar) [reply tests/backup/backup_restore_basic_test.go:1582]: There is another test case written for that approach

File: tests/backup/backup_restore_basic_test.go (hunk 2 above)
Comment (apimpalgaonkar) [reply tests/backup/backup_restore_basic_test.go:1549]: same

**Comment by kshithijiyer-px** [comment]
Issue fixed in this PR https://github.com/portworx/torpedo/pull/1425. Hence closing it.

**Pull request closed without merging**